manage/config-dialog.js

```diff
@@ -286,7 +312,41 @@ function configDialog(style) {
   }
 
   function updateVarOnChange() {
-    this.va.value = this.type !== 'checkbox' ? this.value : this.checked ? '1' : '0';
+    if (this.type === 'range') {
+      this.va.value = Number(this.value);
+      updateRangeCurrentValue(this.va, this.va.value);
+    } else if (this.type === 'number') {
+      const value = Number(this.value);
+      if (clampValue(value, this.va) === value) {
```

Comment:
Good idea, but it should be `String(clampValue(value, this.va)) === this.value`. Or it would match invalid values for some edge cases e.g. the input is empty `''`, has leading zeros `01`, or has meaningless decimal `1.0`.

Reply:
Stringified comparison won't work if the style specifies `1e2` instead of `100`. It's also a valid JSON number.

Reply:
You're right. Users can also input `1e2` directly. Instead of changing it to `String(clampValue(value, this.va)) === this.value`, I think we shouldn't re-render the input while the input is focused:
```diff
diff --git a/manage/config-dialog.js b/manage/config-dialog.js
index efd720b..65734a5 100644
--- a/manage/config-dialog.js
+++ b/manage/config-dialog.js
@@ -363,6 +363,9 @@ function configDialog(style) {
 
   function renderValues(varsToRender = vars) {
     for (const va of varsToRender) {
+      if (va.input === document.activeElement) {
+        continue;
+      }
       const value = isDefault(va) ? va.default : va.value;
       if (va.type === 'color') {
         va.input.style.backgroundColor = value;
```

Reply:
Added this last bit of code to ignore the input while active; but it looks a little weird to have an empty input, set to zero in the background, and not show anything. The update won't update until another variable is changed.

Reply:
Without e595f23, invalid values are also left in the input until another variable is changed. Maybe it is a good idea to update the input to match `input.va.value` when blurred?

Reply:
Ok, that works better.